node-notes-cli/app.js

```diff
@@ -0,0 +1,113 @@
+import * as fs from 'node:fs/promises';
+
+const cliArgs = process.argv.slice(2);
+
+/* This function logs all the notes from the JSON file to the console.
+   I did not make it async since we use no asynchronous functions. */
+function readNotes(data) {
+  try {
+    const { notes } = data;
+    for (const item in notes) {
+      console.log(`${item}: ${notes[item]}`);
+    }
+  } catch (error) {
+    console.log('ERROR: ', error);
+  }
+}
+
+/* This function adds a new note to the json object. I chose to update
+   the nextId by a random amount b/c it's more fun :) */
+
+async function addNote(data, userNote) {
+  try {
+    const { nextId, notes } = data;
+    notes[nextId] = userNote;
+    const newId = nextId + Math.floor(Math.random() * 5 + 1);
+    data.nextId = newId;
+    await fs.writeFile('./data.json', JSON.stringify(data));
```

Comment:
You have this line of code in here several times. It might be a good idea to extract it into a separate function (which makes it easier to maintain).